### PR TITLE
JDK-8290845: Consider an alternative item separator for multi-item option values

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a medium simple update to prefer the use of comma (`,`) instead of colon (`:`) in certain options taking a list of values.    Options taking a search path (like `--class-path`, `--source-path`, etc) are **not** affected.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8290845](https://bugs.openjdk.org/browse/JDK-8290845): Consider an alternative item separator for multi-item option values
 * [JDK-8295858](https://bugs.openjdk.org/browse/JDK-8295858): Consider an alternative item separator for multi-item option values (**CSR**)

### Reviewers
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11178/head:pull/11178` \
`$ git checkout pull/11178`

Update a local copy of the PR: \
`$ git checkout pull/11178` \
`$ git pull https://git.openjdk.org/jdk pull/11178/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11178`

View PR using the GUI difftool: \
`$ git pr show -t 11178`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11178.diff">https://git.openjdk.org/jdk/pull/11178.diff</a>

</details>

### CODE REVIEW

File: test/langtools/jdk/javadoc/tool/CheckManPageOptions.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static void main(String... args) throws Exception {
 
     static final PrintStream out = System.err;
 
-    List<String> MISSING_IN_MAN_PAGE = List.of("--spec-base-url");
+    List<String> MISSING_IN_MAN_PAGE = List.of();
```
Comment (pavelrappo) [inline test/langtools/jdk/javadoc/tool/CheckManPageOptions.java:69]: What's that about? It seems unrelated.

Comment (pavelrappo) [reply test/langtools/jdk/javadoc/tool/CheckManPageOptions.java:69]: It's as if this mainline commit somehow got into this PR: https://github.com/openjdk/jdk/commit/82f9819eaccd091c9d3a0b89979ddc13b1ef761c

Comment (jonathan-gibbons) [reply test/langtools/jdk/javadoc/tool/CheckManPageOptions.java:69]: I'll investigate, but it looks like just a side effect of importing the latest troff man page.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java [hunk 2]
```diff
@@ -471,9 +471,13 @@ private void initTagletManager(Set<List<String>> customTagStrs) {
                     tagletManager.addCustomTag(args.get(1), fileManager);
                     continue;
                 }
-                List<String> tokens = tokenize(args.get(1), TagletManager.SIMPLE_TAGLET_OPT_SEPARATOR, 3);
+                /* Since there are few constraints on the characters in a tag name,
+                 * and real world examples with ':' in the tag name, we cannot simply use
+                 * String.split(regex);  instead, we tokenize the string, allowing
+                 * special characters to be escaped with '\'. */
+                List<String> tokens = tokenize(args.get(1),3);
```
Comment (ExE-Boss) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java:478]: This should have a space:
```suggestion
                List<String> tokens = tokenize(args.get(1), 3);
```

File: test/langtools/jdk/javadoc/tool/CheckManPageOptions.java (hunk 1 above)
Comment (jonathan-gibbons) [reply test/langtools/jdk/javadoc/tool/CheckManPageOptions.java]: It's now been merged in.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java (hunk 2 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java]: Fixed.

**Review by pavelrappo — APPROVED** [review]

**Review by pavelrappo — APPROVED** [review]
Looks good to me.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java [hunk 3]
```diff
@@ -505,18 +507,17 @@ private void initTagletManager(Set<List<String>> customTagStrs) {
...
-     * @param separator the separator char.
+     * @param s         the string to tokenize
      * @param maxTokens the maximum number of tokens returned.  If the
      *                  max is reached, the remaining part of s is appended
      *                  to the end of the last token.
-     * @return an array of tokens.
+     * @return an array of tokens
      */
-    private List<String> tokenize(String s, char separator, int maxTokens) {
+    private List<String> tokenize(String s, int maxTokens) {
```
[the remark this answers is not in the document]
Comment (RogerRiggs) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseConfiguration.java:520]: None of the JDK simple tokenizers supports handling of escape characters.
But a Pattern/Matcher could be used.
